packages/app-builder-lib/src/targets/FlatpakTarget.ts

```diff
@@ -0,0 +1,169 @@
+    })
+
+    await Promise.all(copyIcons)
+  }
+
+  private getFlatpakBuilderOptions(appOutDir: string, stageDir: string, artifactName: string, arch: Arch): { manifest: FlatpakManifest; buildOptions: FlatpakBundlerBuildOptions } {
+    const appIdentifier = this.packager.appInfo.id
+    const { executableName } = this.packager
+    const flatpakArch = toLinuxArchString(arch, "flatpak")
+
+    const manifest: FlatpakManifest = {
+      id: appIdentifier,
+      command: "electron-wrapper",
+      runtime: this.options.runtime || flatpakBuilderDefaults.runtime,
+      runtimeVersion: this.options.runtimeVersion || flatpakBuilderDefaults.runtimeVersion,
+      sdk: this.options.sdk || flatpakBuilderDefaults.sdk,
+      base: this.options.base || flatpakBuilderDefaults.base,
+      baseVersion: this.options.baseVersion || flatpakBuilderDefaults.baseVersion,
+      finishArgs: this.options.finishArgs || flatpakBuilderDefaults.finishArgs,
+      branch: this.options.branch,
+      modules: this.options.modules,
+    }
+
+    const buildOptions: FlatpakBundlerBuildOptions = {
+      baseFlatpakref: `app/${manifest.base}/${flatpakArch}/${manifest.baseVersion}`,
+      runtimeFlatpakref: `runtime/${manifest.runtime}/${flatpakArch}/${manifest.runtimeVersion}`,
+      sdkFlatpakref: `runtime/${manifest.sdk}/${flatpakArch}/${manifest.runtimeVersion}`,
+      arch: flatpakArch as any,
+      bundlePath: path.join(this.outDir, artifactName),
+      files: [[stageDir, "/"], [appOutDir, path.join("/lib", appIdentifier)], ...(this.options.files || [])],
+      symlinks: [[path.join("/lib", appIdentifier, executableName), path.join("/bin", executableName)], ...(this.options.symlinks || [])],
+    }
+
+    return { manifest, buildOptions }
+  }
+
+  private isElectronVersionGreaterOrEqualThan(version: string): boolean {
+    return semver.gte(this.packager.config.electronVersion || "7.0.0", version)
+  }
+}
+
+const flatpakBuilderDefaults: Omit<FlatpakManifest, "id" | "command"> = {
+  runtime: "org.freedesktop.Platform",
+  runtimeVersion: "20.08",
+  sdk: "org.freedesktop.Sdk",
+  base: "org.electronjs.Electron2.BaseApp",
+  baseVersion: "20.08",
+  finishArgs: [
+    // Wayland
+    "--socket=wayland",
+    // X11
+    "--socket=x11",
+    "--share=ipc",
+    // Open GL
+    "--device=dri",
+    // Audio output
+    "--socket=pulseaudio",
+    // Read/write home directory access
+    "--filesystem=home",
+    // Chromium uses a socket in tmp for its singleton check
+    "--env=TMPDIR=/var/tmp",
+    // Allow communication with network
+    "--share=network",
+    // System notifications with libnotify
+    "--talk-name=org.freedesktop.Notifications",
+  ],
+}
+
+function getElectronWrapperScript(executableName: string, isWaylandSupported: boolean): string {
+  return isWaylandSupported
+    ? `#!/bin/sh
+
+if [ "\${XDG_SESSION_TYPE}" == "wayland" ]; then
+    zypak-wrapper "${executableName}" --enable-features=UseOzonePlatform --ozone-platform=wayland "$@"
```

Comment:
Is this a good default yet? I love the idea of Wayland support, but last time I tried it on Chromium itself it still had some rough edges (mostly in terms of font rendering & cursor scaling)...maybe make it opt-in by the packager to encourage use while still avoiding accidental breakage?

Reply:
That's a good point. I've changed it to an opt-in option which is `false` by default.